src/backend/src/services/projects.services.ts

```diff
@@ -391,4 +392,97 @@ export default class ProjectsService {
 
     return;
   }
+  /**
+   * Delete the the project in the database along with all its work packages.
+   * @returns the project that is deleted.
+   */
+  static async deleteProject(user: User, wbsNumber: WbsNumber): Promise<Project> {
+    if (!isProject(wbsNumber)) throw new HttpException(400, `${wbsPipe(wbsNumber)} is not a valid project WBS #!`);
+    if (user.role === Role.GUEST || user.role === Role.LEADERSHIP || user.role === Role.MEMBER) {
+      throw new AccessDeniedException('Guests, Members, and Leadership cannot delete projects');
+    }
+
+    const { carNumber, projectNumber, workPackageNumber } = wbsNumber;
+
+    const project = await prisma.project.findFirst({
+      where: {
+        wbsElement: {
+          carNumber,
+          projectNumber,
+          workPackageNumber
+        }
+      },
+      ...projectQueryArgs
+    });
+
+    if (!project) throw new NotFoundException('Project', wbsPipe(wbsNumber));
+    if (project.wbsElement.dateDeleted) throw new HttpException(400, 'This project has been deleted!');
+
+    const { projectId } = project;
+    const dateDeleted: Date = new Date();
+    const deletedProject = await prisma.project.update({
+      where: {
+        projectId
+      },
+      data: {
```

Comment:
we also need to delete all associated change requests which is a field of wbselement -- shouldn't be too bad but check peyton's endponit just in case to make sure w edon't miss any data